proto_proxy.c

```diff
@@ -2448,18 +2494,18 @@ static void proxy_backend_handler(const int fd, const short which, void *arg) {
                 P_DEBUG("%s: read: %d\n", __func__, read);
                 if (read == 0) {
                     // not connected or error.
-                    _reset_bad_backend(be);
+                    _reset_bad_backend(be, P_BE_FAIL_DISCONNECTED);
                     return;
                 } else if (read == -1) {
                     if (errno == EAGAIN || errno == EWOULDBLOCK) {
                         break; // sit on epoll again.
                     } else {
-                        _reset_bad_backend(be);
+                        _reset_bad_backend(be, P_BE_FAIL_READING);
                         return;
                     }
                 }
             } else if (res == -1) {
-                _reset_bad_backend(be);
+                _reset_bad_backend(be, P_BE_FAIL_PARSING);
```

Comment:
Does this mean the command was improperly formatted?

I'm guessing this will be useful if there's a command received by the proxy and then transformed into a different command(s) via a Lua hook function, and this transformed command is incorrectly formatted. And this could only happen if there's a bug in the Lua function. Am I understanding it correctly?

Reply:
This would be because the remote server sent a response it didn't understand for some reason (any reason why that would happen; lua sent junk or the backend server is broken and sent garbage)